code/modules/vehicles/construction.dm

```diff
@@ -272,3 +273,103 @@
+					return
+				to_chat(user, "<span class='notice'>You start to add treads to [src].</span>")
+				if(do_after(user, 40) && build_stage == 0)
+					if(S.use(6))
+						to_chat(user, "<span class='notice'>You add treads to \the [src].</span>")
+						increase_step("tracked [initial(name)]")
+				return
+
+		if(1)
+			if(istype(W, /obj/item/stock_parts/console_screen))
+				user.drop_item()
+				qdel(W)
+				to_chat(user, "<span class='notice'>You add the lights to \the [src].</span>")
+				increase_step()
+				return
+
+		if(2)
+			if(istype(W, /obj/item/stock_parts/spring))
+				user.drop_item()
+				qdel(W)
+				to_chat(user, "<span class='notice'>You add the control system to \the [src].</span>")
+				increase_step()
+				return
+
+		if(3)
+			if(istype(W, /obj/item/stack/cable_coil))
+				var/obj/item/stack/cable_coil/C = W
+				if (C.get_amount() < 2)
+					to_chat(user, "<span class='warning'>You need two coils of wire to wire [src].</span>")
+					return
+				to_chat(user, "<span class='notice'>You start to wire [src].</span>")
+				if(do_after(user, 40) && build_stage == 3)
+					if(C.use(2))
+						to_chat(user, "<span class='notice'>You wire \the [src].</span>")
+						increase_step("wired [initial(name)]")
+				return
+
+		if(4)
+			if(istype(W, /obj/item/cell))
+				user.drop_item()
+				W.forceMove(src)
+				cell = W
+				to_chat(user, "<span class='notice'>You add the power supply to \the [src].</span>")
+				increase_step("powered [initial(name)]")
+				return
+
+		if(5)
+			if(istype(W, /obj/item/stock_parts/motor))
+				user.drop_item()
+				qdel(W)
+				to_chat(user, "<span class='notice'>You add the motor to \the [src].</span>")
+				increase_step()
+				return
+
+		if(6)
+			if(istype(W, /obj/item/stack/material/plasteel))
+				var/obj/item/stack/material/plasteel/PL = W
+				if (PL.get_amount() < 2)
+					to_chat(user, "<span class='warning'>You need two sheets of plasteel to add reinforcement to \the [src].</span>")
+					return
+				to_chat(user, "<span class='notice'>You start to add reinforcement to [src].</span>")
+				if(do_after(user, 40) && build_stage == 6)
+					if(PL.use(2))
+						to_chat(user, "<span class='notice'>You add reinforcement to \the [src].</span>")
+						increase_step("reinforced [initial(name)]")
+					return
+
+		if(7)
+			if(W.is_wrench() || W.is_screwdriver())
+				playsound(src, W.usesound, 50, 1)
+				to_chat(user, "<span class='notice'>You begin your finishing touches on \the [src].</span>")
+				if(do_after(user, 20) && build_stage == 7)
+					playsound(src, W.usesound, 30, 1)
+					var/obj/vehicle/train/engine/quadbike/snowmobile/built/product = new(src)
+					to_chat(user, "<span class='notice'>You finish \the [product]</span>")
+					product.loc = get_turf(src)
+					product.cell = cell
+					cell.forceMove(product)
+					cell = null
+					user.drop_from_inventory(src)
+					qdel(src)
+				return
```

Comment:
Missing `. = ..()` to handle whacking it with other things?

Reply:
Added at top instead, in keeping with the quad